static/src/javascripts/projects/commercial/modules/cmp/lotame-cmp.js

```diff
@@ -0,0 +1,102 @@
+// @flow
+import { local } from 'lib/storage';
+
+type LotameError = {
+    error: number,
+};
+
+type LotameConsentType = {
+    name: string,
+    consent: boolean,
+};
+
+type LotameClientConsent = {
+    clientId: string,
+    lastupdate: number,
+    types: Array<LotameConsentType>,
+};
+
+type LotameSuccess = {
+    consent: Array<LotameClientConsent>,
+};
+
+const clientId: number = 12666;
+const lotameVendorId: number = 95;
+const lotameConsentKey: string = 'lotameConsent';
+
+const lotameConsentData = (isConsenting: boolean) => ({
+    analytics: isConsenting,
+    crossdevice: isConsenting,
+    datasharing: isConsenting,
+    targeting: isConsenting,
+});
+
+const getLotameConsent = (): number => local.get(lotameConsentKey);
+
+const setLotameConsent = (consent: boolean) =>
+    local.set(lotameConsentKey, consent ? 1 : 0);
+
+const lotameCallback = (isConsenting: boolean) => (
+    data: LotameSuccess | LotameError
+): void => {
+    if ('error' in data) {
+        setLotameConsent(false);
+    } else if ('consent' in data) {
+        setLotameConsent(isConsenting);
+    }
+};
+
+const isConsentingData = (consentData): boolean => {
+    try {
+        const vendorConsents = consentData.vendorConsents;
+        const purposeConsents = consentData.purposeConsents;
+        return (
+            Object.keys(vendorConsents).every(k => vendorConsents[k]) &&
+            Object.keys(purposeConsents).every(k => purposeConsents[k])
+        );
+    } catch (e) {
+        return false;
+    }
+};
+
+const getLotameAdConsentFromCmp = (): Promise<any> =>
```

Comment:
Could this have a more restrictive type?  Probably not but just asking.

Reply:
I wanted the same. However, the types from the `__cmp` object haven't been defined anywhere yet as it seems to have been avoided by the usage of `FlowFixMe`. [See here for example](https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/commercial/modules/cmp/cmp.js#L112).